src/Microsoft.TemplateEngine.Abstractions/GlobalSettings/TemplatePackageData.cs

```diff
@@ -0,0 +1,19 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.TemplateEngine.Abstractions.GlobalSettings
+{
+    public class TemplatePackageData
```

Comment:
If this shouldn't be derived from, can we make this sealed?

Reply:
This class is used for serialization only, I converted it to struct and documented in next commit.

Reply:
In ideal world of C#9, all this properties would have `init;`.
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/init

I think that we can't make setters private and use it as constructor here because Json deserialiser wouldn't like that... 
Long term we will use C#9 and use init here...

Reply:
I made it immutable now.